### PR TITLE
feat: merge discourse-debtcollective-collectives plugin

### PR DESCRIPTION
**Pull request by orlando**
**What:** merge discourse-debtcollective-collectives plugin

**Why:** Closes https://app.asana.com/0/1168997577035609/1179368978953609

**How:**

- Merge front-end code from https://github.com/debtcollective/discourse-debtcollective-collectives.
- Fix issue with lint-staged not processing files with ext `.js.es6` or `.es6`
- Add LICENSE file

### CODE REVIEW

File: javascripts/discourse/templates/components/topic-footer-buttons.hbs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  {{plugin-outlet name="after-topic-footer-main-buttons"
+                  args=(hash topic=topic)
+                  tagName=""
+                  connectorTagName="span"}}
+
+  {{pinned-button pinned=topic.pinned topic=topic}}
+</div>
+
+{{#if showNotificationsButton}}
+{{topic-notifications-button notificationLevel=topic.details.notification_level topic=topic}}
```
Comment (duranmla) [inline javascripts/discourse/templates/components/topic-footer-buttons.hbs:64]: What happen with your formatter? something like this one I think we already installed within your machine:
https://marketplace.visualstudio.com/items?itemName=EmberTooling.prettier-for-handlebars-vscode

File: scss/widgets/dc-collectives-alert.scss [hunk 2]
```diff
@@ -0,0 +1,45 @@
+.alert.alert-collective-alert {
+  align-items: center;
+  background-color: $yellow;
+  border-radius: 4px;
```
Comment (duranmla) [inline scss/widgets/dc-collectives-alert.scss:4]: keep using `rem` as a good practice instead `px`

File: javascripts/discourse/templates/components/dc-collective-alert.hbs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{{#unless hidden}}
+<div class="row">
+  <div class="alert alert-info alert-collective-alert">
+    <span>
+      {{{message}}}
+    </span>
+    {{d-button action=(action "join")
+    class="btn-primary"
+    label="discourse_debtcollective_collectives.join"
```
Comment (duranmla) [inline javascripts/discourse/templates/components/dc-collective-alert.hbs:9]: Are you sure translations works here like that? the way we have been using this is like:
`I18n.t(themePrefix("dc.popup.description"))` where `dc.popup.description` is the label

**Comment by orlando** [comment]
@duranmla here I'm just merging the code from the other plugin, If we want to address other fixes let's make them in other PR

**Pull request merged**